### DISCUSSION TITLE
Curl unable to download error

### DISCUSSION THREAD

**Discussion by Fuzzycc** [JanDeDobbeleer/oh-my-posh#4535] [Troubleshoot] [answered]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

Hello, I was trying to install following the guide for Linux (im on Ubuntu), using curl from snap (curl 8.1.2)
But I keep getting this error:

(Im posting from mobile, so im writing the output log by hand)

Command used:
```
curl -s https://ohmyposh.dev/install.sh | bash -s -- -d ~/bin
```

### Theme

...

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

fish

### Log output

```Shell
installation directory /home/sharbel/bin is not in your $PATH

Installing oh-my-posh for Linux-amd64 in /home/sharbel/bin
Downloading oh-my-posh from https://github.com/JanDeDobbeleer/oh-my-posh/releases/latest/download/posh-linux-amd64
Unable to download executable at https://github.com/JanDeDobbeleer/oh-my-posh/releases/latest/download/posh-linux-amd64
Please validate your curl, connection and/or proxy settings
```

**Comment by JanDeDobbeleer** [comment]
@ChillFire22 not much I can do about that on my end. The curl command `curl -s -f -L https://github.com/JanDeDobbeleer/oh-my-posh/releases/latest/download/posh-linux-amd64 -o /home/sharbel/bin/oh-my-posh` isn't executing correctly.

**Reply by Fuzzycc** [reply]
What do you think I should I do? I managed to manually install the the `install.sh` and the `posh-linux-amd64` , do you think it is possible to manually install?
If nothing works I guess I will look into the homebrew method as mentioned in the installation guide.

Thank you for replying

**Reply by JanDeDobbeleer** [reply, marked as answer]
Manual works, just make sure the binary is executable and lives on your PATH. That's all you need to use it, then follow the setup guide for your shell and start customizing.

**Reply by Fuzzycc** [reply]
Well, I'm a beginner but I'll figure that out eventually I guess.
Thanks for the follow up reply

**Reply by JanDeDobbeleer** [reply]
The install script does the same, you can read through it to see what to do.

**Reply by Fuzzycc** [reply]
Yeah, I read through the script and ended up with a curl (23) error after a cannot open file, permission denied error. So I added sudo before `bash` in the install command and it worked.

It installed except for themes, which gave the same unable to download... themes.zip. I'll look around to see how I can fix this too. Hopefully.

**Reply by Fuzzycc** [reply]
Alright, everything worked fine in the end.

I downloaded and modified the `install.sh` file manually downloading the `themes.zip` file and replacing any relevant variables (namely, theme_dir and zip_file) and commenting out the part that downloads the zip. I also ran the script with sudo, which fixed the download failure of the oh-my-posh executable.

Commenting this for anyone in the future.

Observation:
I had two problems - error downloading the executable, and failing to download the themes.zip after the first error was fixed.
I fixed the first by adding a sudo to the script.
The former, as detailed above for the fix, was (I think) error-ing on me because it replaced cache_dir variable by /root/.cache/oh-my-posh instead of /home/yourname/.cache/oh-my-posh.

**Reply by Maxmo-Felipe** [reply]
Mine was solved by installing it as the root user.
Sometimes the error can persist if you have manually created an oh-my-posh folder in any other directory on the system.
